pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -38,6 +39,39 @@ func NewReconciler(istio *Client) Reconciler {
 	}
 }
 
+func (r *reconciler) ReconcileCSIDriver(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error {
+	log.Info("reconciling istio components for the CSI driver")
+
+	if dynakube == nil {
+		return errors.New("can't reconcile csi driver of nil dynakube")
+	}
+
+	codeModulesURL := dynakube.Status.CodeModules.ImageID
+
+	if !hasCorrectFormat(codeModulesURL) {
+		codeModulesURL = "https://" + codeModulesURL
+	}
+
+	codeModulesHost, err := dtclient.ParseEndpoint(codeModulesURL)
+	if err != nil {
+		return err
+	}
+
+	err = r.reconcileCommunicationHosts(ctx, []dtclient.CommunicationHost{codeModulesHost}, CSIDiverComponent)
+	if err != nil {
+		return errors.WithMessage(err, "error reconciling config for codeModulesImage")
+	}
+
+	log.Info("reconciled istio objects for CSI driver")
+
+	return nil
+}
+
+// if the URL does not have the correct format parsing does not work‚
+func hasCorrectFormat(url string) bool {
+	return strings.HasPrefix(url, "https://")
```

Comment:
what if I like to live dangerously an I have `http://`

Reply:
Good point. I am not sure how I could implement this as usualy the `imageID` that is set comes with no protocol at all. `dtclient.ParseEndpoint` fails when there is no protocol is set, which is why I did it that way. Now I am unsure how I could make this check between `http` and `https`

Do you have any suggestions?

Reply:
you can check if there is a schema or not with the `url` library, and append `https` if nothing is present , but the current logic would just append `https://` to almost anything :D

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3343/commits/c92b4c3055a108823e37d4945a6230151c798548